<?xml version="1.0" encoding="utf-8"?>

<launch>
    <arg name='camera_topic' default='$(optenv BADGR_INPUT_TOPIC)'/>
    <arg name='control_topic' default='$(optenv BADGR_OUTPUT_CONTROL_TOPIC)'/>
    <arg name='output_image_topic' default='$(optenv BADGR_OUTPUT_IMAGE_TOPIC)'/>
    <arg name='velocity_topic' default='$(optenv BADGR_VELOCITY_TOPIC)'/>
    <arg name='control_freq' default='$(optenv BADGR_CONTROL_FREQ)'/>
    <arg name='batches' default='$(optenv BADGR_SAMPLE_BATCHES)'/>
    <arg name='horizon' default='$(optenv BADGR_PLANNING_HORIZON)'/>
    <arg name='initial_velocity' default='$(optenv BADGR_INITIAL_VELOCITY)'/>
    <arg name='initial_steer' default='$(optenv BADGR_INITIAL_STEERING_ANGLE)'/>
    <arg name='gamma' default='$(optenv BADGR_UPDATE_WEIGHTING)'/>
    <arg name='velocity_variance' default='$(optenv BADGR_SAMPLE_VELOCITY_VARIANCES)'/>
    <arg name='steering_variance' default='$(optenv BADGR_SAMPLE_STEERING_VARIANCES)'/>
    <arg name='goal_gain' default='$(optenv BADGR_GOAL_GAIN)'/>
    <arg name='action_gain' default='$(optenv BADGR_ACTION_GAIN)'/>
    <arg name='wheel_base' default='$(optenv BADGR_WHEEL_BASE)'/>
    <arg name='model_name' default='$(optenv BADGR_MODEL_NAME)'/>

  <!-- <arg name="node_start_delay" default="1.0" />   -->
  <!-- launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' " -->
  <!-- <node name="carla_environment" pkg="capra_ros_badgr" type="Carla_Run_Herdr.py" output="screen" /> -->

  <node name="herdr_control_policy" pkg="capra_ros_badgr" type="Real_Life_Test.py" output="screen">
    <param name="camera_topic" value="$(arg camera_topic)" unless="$(eval camera_topic == '')"/>
    <param name="control_topic" value="$(arg control_topic)" unless="$(eval control_topic == '')"/>
    <param name="output_image_topic" value="$(arg output_image_topic)" unless="$(eval output_image_topic == '')"/>
    <param name="velocity_topic" value="$(arg velocity_topic)" unless="$(eval velocity_topic == '')"/>
    <param name="control_freq" value="$(arg control_freq)" unless="$(eval control_freq == '')"/>
    <param name="batches" value="$(arg batches)" unless="$(eval batches == '')"/>
    <param name="horizon" value="$(arg horizon)" unless="$(eval horizon == '')"/>
    <param name="initial_velocity" value="$(arg initial_velocity)" unless="$(eval initial_velocity == '')"/>
    <param name="initial_steer" value="$(arg initial_steer)" unless="$(eval initial_steer == '')"/>
    <param name="gamma" value="$(arg gamma)" unless="$(eval gamma == '')"/>
    <param name="velocity_variance" value="$(arg velocity_variance)" unless="$(eval velocity_variance == '')"/>
    <param name="steering_variance" value="$(arg steering_variance)" unless="$(eval steering_variance == '')"/>
    <param name="goal_gain" value="$(arg goal_gain)" unless="$(eval goal_gain == '')"/>
    <param name="action_gain" value="$(arg action_gain)" unless="$(eval action_gain == '')"/>
    <param name="wheel_base" value="$(arg wheel_base)" unless="$(eval wheel_base == '')"/>
    <param name="model_name" value="$(arg model_name)" unless="$(eval model_name == '')"/>
    <remap from="input_image" to="$(arg camera_topic)"/>
    <remap from="output_image" to="$(arg output_image_topic)"/>
    <remap from="output_control" to="$(arg control_topic)"/>
    <remap from="input_vel" to="$(arg velocity_topic)"/>
  </node>


  
  <!-- <node name="subscriber" pkg="capra_ros_badgr" type="subscriber.py" output="screen"/> -->
  <!-- <node name="talker_node" pkg="capra_ros_badgr" type="Publisher.py" output="screen"/> -->
</launch>
